xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -151,65 +195,11 @@ double GetDropoutRateFromHlo(HloInstruction* dropout) {
   return (1.0 - (1.0 / *dropout_rate_inv));
 }
 
-HloInstruction* GetBiasFromBmm(HloInstruction* bmm) {
-  return bmm->mutable_operand(2);
-}
-
-// Give this instruction a more useful name than "custom-call.42".
-Status SetName(HloModule* module, HloInstruction* fmha) {
-  if (fmha->custom_call_target() == kCudnnfMHABmmBmmCallTarget) {
-    module->SetAndUniquifyInstrName(fmha, "fmha-bmm-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() == kCudnnfMHASoftmaxDropoutCallTarget) {
-    module->SetAndUniquifyInstrName(fmha, "fmha-bmm-softmax-dropout-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() == kCudnnfMHAScaleMaskSoftmaxCallTarget) {
-    module->SetAndUniquifyInstrName(fmha, "fmha-bmm-scale-mask-softmax-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() ==
-      kCudnnfMHAScaleMaskSoftmaxDropoutCallTarget) {
-    module->SetAndUniquifyInstrName(fmha,
-                                    "fmha-bmm-scale-mask-softmax-dropout-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() == kCudnnfMHAScaleBiasMaskSoftmaxCallTarget) {
-    module->SetAndUniquifyInstrName(fmha,
-                                    "fmha-bmm-scale-bias-mask-softmax-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() ==
-      kCudnnfMHAScaleBiasMaskSoftmaxDropoutCallTarget) {
-    module->SetAndUniquifyInstrName(
-        fmha, "fmha-bmm-scale-bias-mask-softmax-dropout-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() == kCudnnfMHASoftmaxCallTarget) {
-    module->SetAndUniquifyInstrName(fmha, "fmha-bmm-softmax-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() == kCudnnfMHAScaleBiasSoftmaxCallTarget) {
-    module->SetAndUniquifyInstrName(fmha, "fmha-bmm-scale-bias-softmax-bmm");
-    return OkStatus();
-  }
-  if (fmha->custom_call_target() ==
-      kCudnnfMHAScaleBiasSoftmaxDropoutCallTarget) {
-    module->SetAndUniquifyInstrName(fmha,
-                                    "fmha-bmm-scale-bias-softmax-dropout-bmm");
-    return OkStatus();
-  }
-
-  return InternalError(
-      "Found invalid FMHA custom-call target while setting custom-call name");
-}
-
 bool IsComputeCapabilityAndCudnnSupported(
     stream_executor::CudaComputeCapability cc,
     stream_executor::dnn::VersionInfo cudnn_version,
-    stream_executor::StreamExecutor* stream_exec) {
-  // return true;
+    stream_executor::StreamExecutor* stream_exec,
+    stream_executor::dnn::VersionInfo supported_cudnn_version) {
```

Comment:
supported_cudnn_version was added since we have different training features available in different versions of cudnn. for version 8.8.0 - 8.9.1, cudnn only supports additive bias and mask. For 8.9.1+ cudnn supports any type of mask. We use this utility function to decide what we would rewrite based on cudnn version.
The previous
`// return true;`
was left there by mistake though..